src/frontend/sharedComponents/MapView.js

```diff
@@ -237,11 +257,31 @@ class MapView extends React.Component<Props, State> {
   };
 
   handleLocationPress = () => {
-    const { location } = this.props;
+    const { location, styleURL } = this.props;
     if (!(location.provider && location.provider.locationServicesEnabled))
       // TODO: Show alert for the user here so they know why it does not work
       return;
-    this.setState(state => ({ following: !state.following }));
+    // Sorry about this mess - when the user is interacting with the map we
+    // don't setState, because it would be too slow, so we use this.zoomRef to
+    // track the current map zoom. This means both this.zoomRef and setState
+    // need to be called to update the zoom of the map. This should be fixed
+    // with a better approach, because it can lead to bugs. For now, this code
+    // just ensures that the map zooms in when the user clicks the "locate me"
+    // button.
+    this.setState(state => {
+      const newZoom = (this.zoomRef =
+        styleURL === "error"
+          ? this.zoomRef < 4
```

Comment:
Math.max(4, this.zoomRef) might be more readable

Reply:
Thanks @okdistribute, problem with open source coding late on Friday, everyone sees your dumb code :P